doc/userguide/upgrade/8.0-dns-logging-changes.rst

```diff
@@ -0,0 +1,260 @@
+:orphan: Document not referenced in a toctree, so add this.
+
+DNS EVE Logging Changes for 8.0
+===============================
+
+Suricata 8.0 modifies the DNS logging in ``dns`` and ``alert`` records
+to a version ``3`` logging format. These changes address a lack of
+fidelity in alerts for DNS responses, as well as unify the format of
+the ``dns`` object accross ``dns`` and ``alert`` objects.
+
+Ticket: https://redmine.openinfosecfoundation.org/issues/6281
+
+The changes are summarized below:
+
+* DNS requests now have a type of ``request`` instead of ``query``.
+
+* DNS responses now have a type of ``response`` instead of ``answer``.
+
+* DNS requests will now log the queries in an array instead of logging
+  multiple request events in the case where the request contained
+  multiple queries. This was already done for DNS requests logged as
+  part of an ``alert``.
+
+  .. list-table::
+     :widths: 50 50
+     :header-rows: 1
+
+     * - 7.0
+       - 8.0
+     * - .. code-block::
+
+             {
+               "event_type": "dns",
+               "dns": {
+                 "type": "query",
+                 "id": 0,
+                 "rrname": "www.suricata.io",
+                 "rrtype": "A",
+                 "tx_id": 0,
+                 "opcode": 0
+               }
+             }
+
+       - .. code-block::
+
+             {
+               "event_type": "dns",
+               "dns": {
+                 "version": 3,
+                 "type": "request",
+                 "tx_id": 0,
+                 "id": 0,
+                 "flags": "100",
+                 "rd": true,
+                 "opcode": 0,
+                 "rcode": "NOERROR",
+                 "queries": [
+                   {
+                     "rrname": "www.suricata.io",
+                     "rrtype": "A"
+                   }
+                 ]
+               }
+             }
+
+* DNS responses now log the queries in a ``queries`` array instead of
+  logging the first ``rrname`` and ``rrtype`` directly in the ``dns``
+  object.
+
+  .. list-table::
+     :header-rows: 1
+
+     * - 7.0
+       - 8.0
+     * - .. code-block::
+
+           {
+             "event_type": "dns",
+             "dns": {
+               "version": 2,
+               "type": "answer",
+               "id": 0,
+               "flags": "8180",
```

Comment:
Any thoughts on if flags would look better as:
```
"flags": ["qr", "rd", "ra"],
```

Reply:
Looks better as ` ["qr", "rd", "ra"],`  than `8180` as a string ?

Reply:
We'd probably want to keep the raw value (`8180` in this case).  The array would replace `"rd": true` and friends.

Reply:
ok as well

Reply:
How would that affect things like elastic and splunk wrt querying for them?

Reply:
Currently you'd have to do something like:
```dns.flags.rd:true dns.flags.z:true```
with this change:
```dns.flags:rd dns.flags:z```

Reply:
Makes sense, thanks